pgxn/remotexact/rwset.c

```diff
@@ -53,12 +54,18 @@ void
 RWSetDecode(RWSet *rwset, StringInfo msg)
 {
 	int			read_set_len;
+	int			num_rels;
 	int			consumed = 0;
+	int			num_rels_received = 0;
 	int			prev_cursor;
 
 	decode_header(rwset, msg);
 
 	read_set_len = pq_getmsgint(msg, 4);
+	num_rels = pq_getmsgint(msg, 4);
+
+	// Allocate relations based on number of relations expected.
+	rwset->relations = alloc_relations(rwset);
```

Comment:
I think you need to set `n_relations` in `rwset` with `num_rels` here

Reply:
Done. I am no longer using num_rels, directly setting n_relations.